mlrun/model_monitoring/db/tsdb/tdengine/tdengine_connector.py

```diff
@@ -260,13 +326,65 @@ def read_predictions(
         start: datetime,
         end: datetime,
         aggregation_window: typing.Optional[str] = None,
+        limit: typing.Optional[int] = None,
     ) -> typing.Union[
         mm_schemas.ModelEndpointMonitoringMetricValues,
         mm_schemas.ModelEndpointMonitoringMetricNoData,
     ]:
-        raise NotImplementedError
+        if not aggregation_window:
+            logger.warning(
+                "Aggregation window is not provided, defaulting to 10 minute."
+            )
+            aggregation_window = "10m"
+
+        df = self.get_records(
+            table=mm_schemas.TDEngineSuperTables.PREDICTIONS,
+            start=start,
+            end=end,
+            columns=["latency"],
+            filter_query=f"endpoint_id='{endpoint_id}'",
+            agg=["count"],
+            interval=aggregation_window,
+            limit=limit,
+        )
+
+        full_name = mlrun.model_monitoring.helpers.get_invocations_fqn(self.project)
+
+        if df.empty:
+            return mm_schemas.ModelEndpointMonitoringMetricNoData(
+                full_name=full_name,
+                type=mm_schemas.ModelEndpointMonitoringMetricType.METRIC,
+            )
+
+        df["_wend"] = pd.to_datetime(df["_wend"])
+        df.set_index("_wend", inplace=True)
+        return mm_schemas.ModelEndpointMonitoringMetricValues(
+            full_name=full_name,
+            values=list(
+                zip(
+                    df.index,
+                    df["count(latency)"],
```

Comment:
Is this correct also in TDEngine? Is that what the query actually returns?

Reply:
yes